### ISSUE TITLE
Adding a disown() method to kj::Own<T>?

### ISSUE THREAD

**Issue by jasnell** [capnproto/capnproto#2089]
One of the current capabilities of `std::unique_ptr<T>` that is not replicated in `kj::Own<T>` is the ability to surrender ownership of the underlying thing. There are obviously some concerns with that if the `kj::Own<T>` is using a custom destroyer but I think in general having a `own.disown()` method that releases the underlying pointer would be helpful. There's at least one case in the workerd codebase where this would be useful. (https://github.com/cloudflare/workerd/pull/2325#discussion_r1651678832)

```
template <typename T>
class Own {
 // ...

  T* disown();
}
```

/cc @mikea

**Comment by kentonv** [comment]
Maybe it should require that you pass in the disposer, to prove you know how this thing is meant to be disposed? If you pass in the wrong disposer, you'd get an exception (rather than a memory leak or UB).

**Comment by jasnell** [comment]
> Maybe it should require that you pass in the disposer...

Good idea.

**Comment by kentonv** [comment]
Implemented in #2092.
